### PR TITLE
feat: Implement comprehensive game save/load functionality with auto-generated filenames

### PR DESCRIPTION
**Pull request by philoz3231**
🎯 Overview
This PR implements a comprehensive game save/load system that allows players to save their poker game state at any point and resume from where they left off. The implementation includes automatic filename generation, smart loading of the most recent save, and robust error handling.

✨ Key Features
- Auto-generated filenames: Save files are automatically named with timestamps (save_YYYYMMDD_HHMMSS.json)
- Smart loading: --load flag automatically loads the most recent save file
- Complete state preservation: All game state including eliminated players, betting rounds, and community cards
- User-friendly CLI: Intuitive command-line interface with helpful error messages
- Comprehensive testing: Full test coverage including edge cases and eliminated player scenarios

🔧 Technical Implementation
- JSON serialization: Clean, readable save file format
- PlayerStatus handling: Correctly saves and loads eliminated player states
- File management: Automatic directory creation and file validation
- Error recovery: Graceful handling of corrupted or missing save files

📝 Usage Examples
```bash
# Start a new game
go run main.go

# Save during gameplay (press 's' key)
# Automatically creates: save_20240115_103000.json

# Load the most recent save
go run main.go --load

# Load a specific save file
go run main.go --load-file my_save

# Specify custom save directory
go run main.go --save-dir custom_saves
```

🧪 Testing
- ✅ Unit tests for serialization/deserialization
- ✅ File I/O operations testing
- ✅ Error handling scenarios
- ✅ Eliminated player state preservation
- ✅ Integration tests for complete game flow

📋 Changes Made
- Added GameSaveData and PlayerSaveData structs
- Implemented SaveGameToFile() and LoadGameFromFile() functions
- Updated CLI with --load, --load-file, and --save-dir flags
- Enhanced error messages with user guidance
- Added comprehensive test coverage
- Updated documentation and README

🎮 User Experience Improvements
- Persistent success messages: Save confirmation with pause for user acknowledgment
- Intelligent error handling: Clear error messages with recovery suggestions
- Seamless workflow: No need to remember or type filenames
- State consistency: Eliminated players remain eliminated after loading

---

🎯 개요
이 PR은 포커 게임 상태를 언제든지 저장하고 중단된 지점부터 재개할 수 있는 포괄적인 게임 저장/로드 시스템을 구현합니다. 자동 파일명 생성, 최신 저장 파일의 스마트 로딩, 그리고 강력한 에러 처리가 포함되어 있습니다.

✨ 주요 기능
- 자동 파일명 생성: 저장 파일이 타임스탬프로 자동 명명됩니다 (save_YYYYMMDD_HHMMSS.json)
- 스마트 로딩: --load 플래그로 최신 저장 파일을 자동 로드
- 완전한 상태 보존: 퇴장된 플레이어, 베팅 라운드, 커뮤니티 카드를 포함한 모든 게임 상태
- 사용자 친화적 CLI: 도움이 되는 에러 메시지와 함께 직관적인 명령행 인터페이스
- 포괄적인 테스트: 엣지 케이스와 퇴장된 플레이어 시나리오를 포함한 전체 테스트 커버리지

🔧 기술적 구현
- JSON 직렬화: 깔끔하고 읽기 쉬운 저장 파일 형식
- PlayerStatus 처리: 퇴장된 플레이어 상태를 정확히 저장하고 로드
- 파일 관리: 자동 디렉토리 생성 및 파일 검증
- 에러 복구: 손상되었거나 누락된 저장 파일의 우아한 처리

📝 사용 예시
```bash
# Start a new game
go run main.go

# Save during gameplay (press 's' key)
# Automatically creates: save_20240115_103000.json

# Load the most recent save
go run main.go --load

# Load a specific save file
go run main.go --load-file my_save

# Specify custom save directory
go run main.go --save-dir custom_saves
```

🧪 테스트
- ✅ 직렬화/역직렬화 단위 테스트
- ✅ 파일 I/O 작업 테스트
- ✅ 에러 처리 시나리오
- ✅ 퇴장된 플레이어 상태 보존
- ✅ 전체 게임 플로우 통합 테스트

📋 변경 사항
- GameSaveData와 PlayerSaveData 구조체 추가
- SaveGameToFile()과 LoadGameFromFile() 함수 구현
- --load, --load-file, --save-dir 플래그로 CLI 업데이트
- 사용자 가이드와 함께 에러 메시지 향상
- 포괄적인 테스트 커버리지 추가
- 문서화 및 README 업데이트

🎮 사용자 경험 개선
- 지속적인 성공 메시지: 사용자 확인을 위한 일시정지와 함께 저장 확인
- 지능적인 에러 처리: 복구 제안과 함께 명확한 에러 메시지
- 원활한 워크플로우: 파일명을 기억하거나 입력할 필요 없음
- 상태 일관성: 로드 후에도 퇴장된 플레이어는 퇴장 상태 유지

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -89,6 +89,12 @@ The application accepts the following flags:
 | `--blind-up`     | `int`    | `2`      | The number of hands for blinds to increase. `0` disables blind-ups.         |
 | `--dev`          | `bool`   | `false`  | Enables development mode for verbose logging.                               |
 | `--outs`         | `bool`   | `false`  | Shows hand outs for the human player.                                       |
+| `--load`, `-l`   | `bool`   | `false`  | Load the most recent saved game.                                            |
+| `--load-file`    | `string` | `""`     | Load a specific saved game file.                                            |
+| `--save-dir`     | `string` | `"saves"`| Directory to store save files.                                             |
+| `--initial-chips`| `int`    | `300000` | Initial chips for each player.                                              |
+| `--small-blind`  | `int`    | `500`    | Small blind amount.                                                         |
+| `--big-blind`    | `int`    | `1000`   | Big blind amount.                                                           |
```
Comment (philipjkim) [inline README.md:97]: 보통 BB 는 SB 의 2배가 통상적인 룰이니 `--small-blind` 만 두거나 `--big-blind` (단 BB 는 짝수값이어야 함) 만 둬도 괜찮을 것 같네요~

File: README.md [hunk 2]
```diff
@@ -103,10 +109,50 @@ go run main.go --rule pls
...
+- Type `q` to quit the game
+
+#### Betting Actions
+During your turn, you can choose from:
+- `f` - Fold (forfeit your hand)
+- `c` - Call (match the current bet)
+- `r` - Raise (increase the bet)
+- `k` - Check (pass without betting, when no bet is required)
+- `b` - Bet (make the first bet in a round)
+- `s` - Save (save the current game state)
```
Comment (philipjkim) [inline README.md:154]: 👍

**Review by philipjkim — APPROVED** [review]
👍👍👍 Fimally!
